plugins/azure/fix_plugin_azure/resource/postgresql.py

```diff
@@ -0,0 +1,641 @@
+    status: Optional[str] = field(default=None, metadata={"description": "The status"})
+    supported_fast_provisioning_editions: Optional[List[AzureFastProvisioningEditionCapability]] = field(default=None, metadata={'description': ''})  # fmt: skip
+    supported_psql_flexible_server_editions: Optional[List[AzureFlexibleServerEditionCapability]] = field(default=None, metadata={'description': ''})  # fmt: skip
+    supported_ha_mode: Optional[List[str]] = field(default=None, metadata={'description': 'Supported high availability mode'})  # fmt: skip
+    supported_hyperscale_node_editions: Optional[List[AzureHyperscaleNodeEditionCapability]] = field(default=None, metadata={'description': ''})  # fmt: skip
+    capability_zone: Optional[str] = field(default=None, metadata={"description": "zone name"})
+    zone_redundant_ha_and_geo_backup_supported: Optional[bool] = field(default=None, metadata={'description': 'A value indicating whether a new server in this region can have geo-backups to paired region.'})  # fmt: skip
+    zone_redundant_ha_supported: Optional[bool] = field(default=None, metadata={'description': 'A value indicating whether a new server in this region can support multi zone HA.'})  # fmt: skip
+    location: Optional[str] = field(default=None, metadata={'description': 'The geo-location where the resource lives'})  # fmt: skip
+
+    def post_process(self, graph_builder: GraphBuilder, source: Json) -> None:
+
+        def collect_editions(edition: AzureFlexibleServerEditionCapability) -> None:
+            server_types = []
+            for version in edition.supported_server_versions or []:
+                for sku in version.supported_vcores or []:
+                    for supported_storage in edition.supported_storage_editions or []:
+                        for storage in supported_storage.supported_storage_mb or []:
+                            server_type = {
+                                "id": f"{edition.name}_{version.name}_{sku.name}_{storage.name}",
+                                "name": f"{edition.name}_{version.name}_{sku.name}_{storage.name}",
+                                "sku": {
+                                    "name": sku.name,
+                                    "tier": edition.name,
+                                    "vCores": sku.v_cores,
+                                    "memoryPerVCoreMb": sku.supported_memory_per_vcore_mb,
+                                },
+                                "storage": {
+                                    "iops": storage.supported_iops,
+                                    "storageSizeGb": (
+                                        storage.storage_size_mb // 1024 if storage.storage_size_mb else 0
+                                    ),
+                                },
+                                "location": self.location,
+                            }
```

Comment:
We create here an arbitrary format which then is parsed in AzurePostgresqlServerType. 
Q1: what is the relationship between capability and server type? Can we get rid of one?
Q2: wouldn't it be simpler to create the server type based on the already parsed capability?

Reply:
We collect capabilities. Then from capabilities we create a `AzurePostgresqlServerType`/`AzureMysqlServerType`. After all works done, we remove capabilities from the graph.

Reply:
Consider removing the capability class and read the related type directly.
This way we do not have an artificial type in our model, nor do we need to maintain 2 mappings.